### ISSUE TITLE
feat: allow user to rename files before downloading

### ISSUE THREAD

**Issue by subrato-pattanaik** [Muhammed-Rahif/Notepad#133]
Hi everyone! 👋 It is the first issue that I am posting in this project. 😊🙈 Really awesome idea 🚀

### Is your feature request related to a problem? Please describe.

In the native notepad, we can rename a file when saving it for the first time. Here, I am not able to rename the file before downloading it because of that in my download directory all the files which I download from you are ending with `Untitled.txt (n)` ( n = no. of the same file downloaded - 1 ). That got me annoyed 😛.

### Describe the solution you'd like

This request is to introduce a feature in your project that allows users (like me) to be able to edit the name of the file which can be triggered either by shortcut key or `In-place input`.

 - With `In-place input`, we can edit the file name at the place where it has been rendered i.e on the top. I found this library [react-inline-input-edit](https://www.npmjs.com/package/react-inline-input-edit) useful to implement this feature.

 - `ctrl + r` shortcut can be used to rename a file and when pressed we will focus the in-place input to edit.

### Addition context

We also need to think of `save as` functionality in our project. Just a random idea for `save as`, we could open a Model with input to edit the filename and two buttons save and cancel. On saving button clicked, we will download the file and on cancel, we'll close the modal.

 Thank you!! 😊

**Comment by Muhammed-Rahif** [comment]
Cool request @subrato-pattanaik! It will be nice if we add rename feature.

My suggestions on this:-
- **Make the filename title (on the left top) content editable. So users can name the file from there! As like the GIF down below:-**

![rename-feat-notepad](https://user-images.githubusercontent.com/73386156/151176248-d21e2fe7-7900-4837-9a0e-a3d3fa1e8e95.GIF)

**Comment by subrato-pattanaik** [comment]
Yes, the design is so neat. I was looking for that feature in this project.

I am also wondering about the keyboard shortcut to rename the filename title. How do we handle the keystroke event for this feature? 

I am thinking about `ctrl + r`, when it is pressed, we will make the input focusable along with the blinking cursor and selection of the existing filename.

What do you suggest about this? Or could we discuss this later once this feature is ready for users?

**Comment by Muhammed-Rahif** [comment]
Adding rename feature on <kbd>ctrl</kbd> + <kbd>r</kbd> shortcut wouldn't be a best practice, because <kbd>ctrl</kbd> + <kbd>r</kbd> used to reload the browser tab in default. Preventing that will create a bad user experience, I think. We can implement this feature on <kbd>F2</kbd>, I think that will be better, IMO.

**Comment by subrato-pattanaik** [comment]
Yes, you're absolutely right. My bad 😅. `F2`  will be better. 

Do we need to rely on external npm lib for Inline input? 

I am interested to work on this feature. This would be my first contribution. 🙈

**Comment by Muhammed-Rahif** [comment]
@subrato-pattanaik I don't think we need to rely on a module, because we can do this by adding attribute `contenteditable="true"` for that title. And when the user press on 'save' button we will grab that value from that title put that value as the filename and download the file. Any other better ideas are welcome!

And all the best for your first contribution 🤗..

**Comment by khushbooag4** [comment]
@subrato-pattanaik Are you working on this
@Muhammed-Rahif I wanna try this.

**Comment by Muhammed-Rahif** [comment]
Hi @subrato-pattanaik @khushbooag4,

I'm writing to let you know that due to unforeseen circumstances, this project has been inactive for the past few months. We apologize for any inconvenience this may have caused.

We've recently switched the technology stack of this project from ReactJS to NextJS. As a result, the issues related to the old stack are no longer applicable.

#### We've deployed the new version of the app to production, and you can access it at this link: **[https://notepad-dusky.vercel.app/](https://notepad-dusky.vercel.app/)**

I would like to take this opportunity to encourage you to actively contribute to the project. If you have any ideas, feature requests, or bug reports, please don't hesitate to create a new issue or submit a pull request on our [GitHub repository](https://github.com/Muhammed-Rahif/Notepad).

Therefore, I'm closing this issue. If you believe that the issue is still relevant to the current version of the project, please feel free to create a new issue and provide details on how to reproduce the problem.

Thank you, @subrato-pattanaik @khushbooag4, for your understanding and for your interest in this project! Let's work together to bring it back to life.

Best regards,  
Muhammed Rahif
